launchpod/angular-launchpod/src/app/app.component.ts

```diff
@@ -1,19 +1,46 @@
-import {BreakpointObserver, Breakpoints} from '@angular/cdk/layout';
-import {Component} from '@angular/core';
-import {Observable} from 'rxjs';
-import {map, shareReplay} from 'rxjs/operators';
+import { BreakpointObserver, Breakpoints } from '@angular/cdk/layout';
+import { Component, OnInit } from '@angular/core';
+import { LoginPopupComponent } from './login-popup/login-popup.component';
+import { FormHandlerService } from './form-handler.service';
+import { Observable } from 'rxjs';
+import { map, shareReplay } from 'rxjs/operators';
+import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
 
 @Component({
   selector: 'app-root',
   templateUrl: './app.component.html',
   styleUrls: ['./app.component.css']
 })
-export class AppComponent {
+export class AppComponent implements OnInit {
+
   title = 'angular-launchpod';
+  loginLink: string;
 
   isHandset$: Observable<boolean> =
       this.breakpointObserver.observe(Breakpoints.Handset)
           .pipe(map(result => result.matches), shareReplay());
 
-  constructor(private breakpointObserver: BreakpointObserver) {}
+  constructor(private breakpointObserver: BreakpointObserver, private formHandlerService: FormHandlerService, public dialog: MatDialog) {}
+
+  ngOnInit(): void {
+    this.getLoginStatus();
+  }
+
+  getLoginStatus() {
+    this.formHandlerService.getLoginData()
+      .subscribe((user) => {
+        if (user.isLoggedIn) {
+          document.getElementById("login-container").innerHTML = user.message;
+        } else {
+          this.loginLink = user.message;
+          this.formHandlerService.sendLoginLink(user.message);
```

Comment:
Should this be 
`this.formHandlerService.sendLoginLink(this.loginLink);` ?

Reply:
When the user is not logged in, the message is only the loginLink. I figured I would send back the message from the user object I got from the servlet. Since I am declaring that the value of loginLink is user.message, I thought it would make more sense to send back the source value in case this.loginLink doesn't update with the user.message informatino.